torch_xla/csrc/ops/softmax_backward.cpp

```diff
@@ -0,0 +1,52 @@
+#include "torch_xla/csrc/ops/softmax_backward.h"
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+#include "torch_xla/csrc/softmax_builder.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::Shape NodeOutputShape(const Value& grad_output, const Value& output,
+                           xla::int64 dim) {
+  auto lower_for_shape_fn =
+      [dim](tensorflow::gtl::ArraySlice<const xla::XlaOp> operands)
+      -> xla::XlaOp {
+    XLA_CHECK_EQ(operands.size(), 2)
+        << "Unexpected number of operands: " << operands.size();
+    return BuildSoftmaxGrad(/*grad_output=*/operands[0],
+                            /*output=*/operands[1], dim);
+  };
+  return InferOutputShape({grad_output.shape(), output.shape()},
```

Comment:
Do we really need to infer it?

Reply:
No we don't, fixed the silliness in LogSoftmaxBackward as well.